### PR TITLE
e2e environment setup

### LINKED ISSUE #5702: Investigate setup and teardown of e2e environment
## Required settings

- [x] Create user with role `customer` → `wp user create customer customer@woocommercecoree2etestsuite.com --user_pass=password --role=customer`
- [x] Set rewrite structure to `postname` → `wp rewrite structure '/%postname%/'`
- [x] Flush rewrite settings → `wp rewrite flush`

## Required theme

- [x] [Storefront](https://wordpress.org/themes/storefront/) → `wp theme activate storefront`

## Required plugins

- [x] [WooCommerce](https://wordpress.org/plugins/woocommerce/)
- [x] [WooCommerce Blocks](https://wordpress.org/plugins/woo-gutenberg-products-block/)
- [x] [WooCommerce Payments](https://wordpress.org/plugins/woocommerce-payments/) or [WooCommerce Stripe Payment Gateway](https://wordpress.org/plugins/woocommerce-gateway-stripe/)

## Required pages

- [x] `Shop` page
- [x] `My Account` page
- [x] `Cart block` page
- [x] `Checkout block` page
- [x] `Terms and Conditions` page
- [x]  `Privacy Policy` page

## Required coupons

- [x] Single-use coupon

## Required products

- [x] Default WooCommerce sample_products.csv

## Required payment options

- [x] Cash on delivery
- [x] Direct bank transfer
- [x] Check payments
- [ ] Express payment

## Required payment shipping

- [x] Normal Shipping
- [x] Free shipping

### PR DESCRIPTION
**Pull request by nielslange**
Fixes #5702 

### Note

This PR aims to set up all options and settings, required by the e2e tests, before running the test suite. The following options and settings must be available/activated:

- User with role `customer`
- Permalinks structure must be set to `postname`
- Storefront theme
- WooCommerce plugin
- WooCommerce Blocks plugin
- WordPress Imports plugin
- WooCommerce sample products
- `Shop` page
- `My Account` page
- `Cart block` page
- `Checkout block` page
- `Terms and Conditions` page
- `Privacy Policy` page
- Payment option `Bank Transfer`
- Payment option `Cash on Delivery`
- Payment option `Cheque`
- Shipping option `Free Shipping`
- Shipping option `Flat Rate Shipping`

### Testing

1. Check out this PR.
2. Start Docker Desktop.
3. Run `npm run wp-env destroy && npm run wp-env start && npm run wp-env:config`.
4. Log into the test environment via http://localhost:8889/ and ensure that all options and settings, as mentioned in the description, are available.

### CODE REVIEW

File: bin/wp-env-config.sh [hunk 1]
```diff
@@ -12,11 +12,51 @@ else
   EXIT_CODE=$?
 fi
 
-## set permalinks for easier wp-json
+# PERMALINK SETTINGS ###########################################################
 wp rewrite structure '/%postname%/'
 wp rewrite flush
+
+# PLUGIN SETTINGS ##############################################################
+wp plugin install woo-gutenberg-products-block
```
Comment (dinhtungdu) [inline bin/wp-env-config.sh:20]: Is this intentional? We have already activated the plugin [here](https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/5823/files#diff-9fd21b09a0144355cfd001e3952767df08b2e04241994470aa1d627f797b171eL6).

Comment (nielslange) [reply bin/wp-env-config.sh:20]: While I probably don't use this PR, `wp plugin install woo-gutenberg-products-block` was intentional. We're want to be able to test the translated cart and checkout block. To accomplish that, we need to fetch the translations from WordPress.org. The problem that we're facing is, that we're using the slug  `woocommerce-gutenberg-products-block` internally, but on WordPress.org, the plugin is using the slug `woo-gutenberg-products-block`. By installing (but not activating) the plugin with the slug `woo-gutenberg-products-block`, I'm able to fetch the corresponding translation files from WordPress.org. 

It's rather a dirty hack, than best practice, though. 🙈

Comment (dinhtungdu) [reply bin/wp-env-config.sh:20]: Thanks for the clarification! It makes sense to me now!

**Comment by nielslange** [comment]
Closing this PR as [the corresponding issue had been closed](https://github.com/woocommerce/woocommerce-gutenberg-products-block/issues/5702).

**Pull request closed without merging**